### ISSUE TITLE
Build 0.4.2 on maven central depends on Snapshot version of pdfbox 2.0.16

### ISSUE THREAD

**Issue by loehra** [vandeseer/easytable#41]
... I stongly suggest to bind the library build to a fixed *release* version of PDF Box, i.e., to replace the version of the PDF box dependency as current:

<version>[2.0.4,)</version>

with e.g., 

<version>2.0.15</version

**Comment by vandeseer** [comment]
Thanks loehra for your comment.

The expression with the open interval should only guarantee a PDFBox version higher than 2.0.4. I assume the reason why you have a snapshot version is because you have a snapshot repository activated in your maven settings. 

You should be able to easily set a specific PDFBox version for your build in order to overcome this issue. But easytable itself does not depend on a snapshot version.

I will think about a specific version for the next release though.

**Comment by manios** [comment]
Hi @vandeseer ,

I think that the solution you proposed is not applicable in all cases. I encountered the same problem in a central CI tool (Jenkins, Bamboo, etc) where I did not have access to the `settings.xml` in order to switch snapshots to `false`. Long story short, I ended up including `pdfbox` itself and excluding it from `easytable`:

```xml
<dependency>
    <groupId>org.apache.pdfbox</groupId>
    <artifactId>pdfbox-tools</artifactId>
    <version>2.0.15</version>
</dependency>
<dependency>
    <groupId>com.github.vandeseer</groupId>
    <artifactId>easytable</artifactId>
    <version>0.4.2</version>
    <exclusions>
        <exclusion>
            <groupId>org.apache.pdfbox</groupId>
            <artifactId>pdfbox</artifactId>
        </exclusion>
    </exclusions>
</dependency>
```

Before doing that I have also tried (according to StackOverflow [#53741982](https://stackoverflow.com/questions/53741982/how-to-ignore-snapshot-dependency-during-maven-build)):

```
mvn versions:resolve-ranges -Dincludes=org.apache.pdfbox.*
```

and also tried (according to StackOverflow [#8415590](https://stackoverflow.com/a/8415590)):

```
mvn clean package --no-snapshot-updates
```

with no result.

I hope @loehra this helps you.

Best regards,  
Christos

**Comment by loehra** [comment]
Christos,
thanks for the reply. Indeed, excluding from easytable and adding "some" version manually did the trick. 
I still don't think that using those ranges is a good thing to do in the first place. Two Maven runs with no change to the poms or source code might end up in a different result, e.g., if a new PDF box version had been released in between. It jeopardizes the stability of ones builds.
I personally consider the excluding/including just as a workaround ... 
Cheers,
Andreas

**Comment by vandeseer** [comment]
Thanks @loehra and thanks @manios for sharing your issues and thoughts. 

I think you are right, there should be a fix release version of PDFbox which easytable should depend upon. I will change that soon (not before next week though since I am traveling right now), and will hence reopen this issue.

**Comment by vandeseer** [comment]
I fixed it in the `develop` branch now and within the next days I will also release (I guess latest next weekend).

**Comment by vandeseer** [comment]
Released it (version 0.4.3, depends on PDFBox 2.0.15).
